provider/linode.go

```diff
@@ -90,7 +90,9 @@ func NewLinodeProvider(domainFilter DomainFilter, dryRun bool) (*LinodeProvider,
 		},
 	}
 
+	ua := fmt.Sprintf("ExternalDNS/%s linodego/%s", appVersion, linodego.Version)
 	linodeClient := linodego.NewClient(oauth2Client)
+	linodeClient.SetUserAgent(ua)
```

Comment:
nit: if I see it correctly you don't need to initiate a new var because you're just using it once. just pass it directly in the SetUserAgent()